### DISCUSSION TITLE
npm installation error?

### DISCUSSION THREAD

**Discussion by ghost** [uNetworking/uWebSockets.js#406] [Q&A] [answered]
```
[root@localhost webrtc]# npm install uNetworking/uWebSockets.js#v18.8.0
npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno ENOENT
npm ERR! enoent Error while executing:
npm ERR! enoent undefined ls-remote -h -t ssh://git@github.com/uNetworking/uWebSockets.js.git
npm ERR! enoent
npm ERR! enoent
npm ERR! enoent spawn git ENOENT
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-12-16T01_06_35_073Z-debug.log
```

![image](https://user-images.githubusercontent.com/47580398/102291453-61e6d600-3f00-11eb-815b-28cc15f32171.png)

**Comment by hst-m** [comment, marked as answer]
looks like you need to install git. make sure it is in your Path https://stackoverflow.com/a/56990788

**Reply by ghost** [reply]
It appears it was the system itself. I had everything installed. I was running this on CentOS 8. I re-built the image, re-installed everything, then npm worked. It may be something quirky with hyper-v and centos.

**Comment by ghost** [comment]
I do have git installed.

**Comment by ghost** [comment]
i can do a git clone no, issues. just running npm it doesnt.

**Comment by ghost** [comment]
You can report this to NPM because that is a shitty error message for their client messing up finding git.

**Comment by ghost** [comment]
Also - if NPM depends on git, it should have git as a dependency when installing. But I am entirely fed up with NPM so I do not care.
